rust/routee-compass-core/src/model/traversal/default/speed_traversal_model.rs

```diff
@@ -263,4 +297,79 @@ mod tests {
         // approx_eq(result.updated_state[1].into(), expected, 0.001);
         approx_eq(state[1].into(), expected, 0.001);
     }
+
+    #[test]
```

Comment:
If I'm right about my other comment, then maybe we could also add testing where the limit unit is different

Reply:
Yeah that's a good idea, I'll add that